File: ProyectoMysql/Api/LogisticStorage/LogisticStorage.EntityLayer/PersonaNaturalEntity.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LogisticStorage.EntityLayer
{
    public partial class EntidadEntity : BaseEntityObject
    {

        public EntidadEntity()
        {
            EntidadId = GetInt32Default();
            Nombres = GetStringDefault();
            ApellidoPaterno = GetStringDefault();
            ApellidoMaterno = GetStringDefault();
            FechaNacimiento = DateTime.Now;
            Direccion = GetStringDefault();
            Telefono = GetStringDefault();
            Correo = GetStringDefault();
            SexoId = GetInt32Default();
            EstadoCivilId = GetInt32Default();
            TipoDocumentoIdentidadId = GetInt32Default();
            NumDocumento = GetStringDefault();
            UbigeoId = GetInt32Default();
            FechaRegistro = DateTime.Now;
            CodUsuario = GetStringDefault();
            EstadoRegistro = GetBooleanDefault();

        }

        public Int32 EntidadId { get; set; }
        public Int32 TipoDocumentoIdentidadId { get; set; }
        public String NumDocumento { get; set; }
        public String Nombres { get; set; }
        public String ApellidoPaterno { get; set; }
        public String ApellidoMaterno { get; set; }
        public DateTime FechaNacimiento { get; set; }
        public Int32 UbigeoId { get; set; }
        public String Direccion { get; set; }
        public String Telefono { get; set; }
        public String Correo { get; set; }
        public Int32 SexoId { get; set; }
        public Int32 EstadoCivilId { get; set; }
        public String CodUsuario { get; set; }
        public DateTime FechaRegistro { get; set; }
        public Boolean EstadoRegistro { get; set; }
        public String NombreComercial { get; set; }

    }
}

File: Proyecto/server/EntityLayer/PersonaNaturalEntity.py
from datetime import datetime
from pydantic import BaseModel
from Utilidades.Enumerado.ProcessActionEnum import ProcessActionEnum

class PersonaNaturalSaveModel(BaseModel):
    PersonaNaturalId: int 
    TipoDocumentoIdentidadId: int 
    NumDocumento: str 
    Nombres: str 
    ApellidoPaterno: str 
    ApellidoMaterno: str 
    FechaNacimiento: datetime 
    UbigeoId: int 
    Direccion: str 
    Telefono: str 
    Correo: str 
    SexoId: int 
    EstadoCivilId: int 
    FechaRegistro: datetime 
    CodUsuario: str 
    EstadoRegistro: bool 
    Action: ProcessActionEnum

class PersonaNaturalItemModel:
    PersonaNaturalId: int 
    TipoDocumentoIdentidadId: int 
    NumDocumento: str 
    Nombres: str 
    ApellidoPaterno: str 
    ApellidoMaterno: str 
    FechaNacimiento: datetime 
    UbigeoId: int 
    Direccion: str 
    Telefono: str 
    Correo: str 
    SexoId: int 
    EstadoCivilId: int 
    FechaRegistro: datetime 
    CodUsuario: str 
    EstadoRegistro: bool 

    def Cargar(_DB):
        c =  PersonaNaturalItemModel()
        c.PersonaNaturalId = _DB["PersonaNaturalId"] 
        c.TipoDocumentoIdentidadId = _DB["TipoDocumentoIdentidadId"] 
        c.NumDocumento = _DB["NumDocumento"] 
        c.Nombres = _DB["Nombres"] 
        c.ApellidoPaterno = _DB["ApellidoPaterno"] 
        c.ApellidoMaterno = _DB["ApellidoMaterno"] 
        c.FechaNacimiento = _DB["FechaNacimiento"] 
        c.UbigeoId = _DB["UbigeoId"] 
        c.Direccion = _DB["Direccion"] 
        c.Telefono = _DB["Telefono"] 
        c.Correo = _DB["Correo"] 
        c.SexoId = _DB["SexoId"] 
        c.EstadoCivilId = _DB["EstadoCivilId"] 
        c.FechaRegistro = _DB["FechaRegistro"] 
        c.CodUsuario = _DB["CodUsuario"] 
        c.EstadoRegistro = bool(ord(_DB["EstadoRegistro"])) 
        return c

    def CargarMain(_DB):
        c =  PersonaNaturalItemModel()
        c.PersonaNaturalId = _DB["PersonaNaturalId"] 
        c.NumDocumento = _DB["NumDocumento"] 
        c.Nombres = _DB["Nombres"] 
        c.ApellidoPaterno = _DB["ApellidoPaterno"] 
        c.ApellidoMaterno = _DB["ApellidoMaterno"] 
        c.FechaRegistro = _DB["FechaRegistro"] 
        c.CodUsuario = _DB["CodUsuario"] 
        c.NomDocumento = _DB["NomDocumento"] 
        return c

    def CargarBuscar(_DB):
        c =  PersonaNaturalItemModel()
        c.PersonaNaturalId = _DB["PersonaNaturalId"] 
        c.TipoDocumentoIdentidadId = _DB["TipoDocumentoIdentidadId"]
        c.NumDocumento = _DB["NumDocumento"] 
        c.Nombres = _DB["Nombres"] 
        c.ApellidoPaterno = _DB["ApellidoPaterno"] 
        c.ApellidoMaterno = _DB["ApellidoMaterno"] 
        return c
